lib/src/widgets/apple_pay.dart

```diff
@@ -1,53 +1,93 @@
+class _ApplePayState extends State<ApplePay> {
+  late Future<PaymentConfiguration> _paymentConfigurationFuture;
+  String _merchantName = "";
 
-  void onApplePayError(error) {
-    onPaymentResult(PaymentCanceledError());
+  @override
+  void initState() {
+    super.initState();
+    _paymentConfigurationFuture = _getPaymentConfigurationFromAsset();
+    _setMerchantName();
   }
 
-  void onApplePayResult(paymentResult) async {
-    final token = paymentResult['token'];
-    final source = ApplePayPaymentRequestSource(token);
-    final paymentRequest = PaymentRequest(config, source);
+  Future<void> _setMerchantName() async {
+    String merchantName =
+        await ApplePayUtils.getMerchantName(widget.paymentProfilePath);
+    if (mounted) {
+      setState(() {
+        _merchantName = merchantName;
+      });
+    }
+  }
 
-    final result = await Moyasar.pay(
-        apiKey: config.publishableApiKey, paymentRequest: paymentRequest);
+  void _onApplePayError(Object? error) {
+    widget.onApplePayError?.call(error);
+  }
 
-    onPaymentResult(result);
+  Future<void> _onApplePayResult(Map<String, dynamic> paymentResult) async {
+    try {
+      final String token = paymentResult['token'];
+      widget.onPaymentResult(token);
+    } catch (e) {
+      _onApplePayError(e);
+    }
   }
 
   @override
   Widget build(BuildContext context) {
-    return ApplePayButton(
-      paymentConfigurationAsset: 'default_payment_profile_apple_pay.json',
-      paymentItems: [
-        PaymentItem(
-          label: '',
-          amount: (config.amount / 100).toStringAsFixed(2),
-        )
-      ],
-      type: ApplePayButtonType.inStore,
-      onPaymentResult: onApplePayResult,
-      width: MediaQuery.of(context).size.width,
-      height: 40,
-      onError: onApplePayError,
-      loadingIndicator: const Center(
-        child: CircularProgressIndicator(),
-      ),
+    return FutureBuilder<PaymentConfiguration>(
+      future: _paymentConfigurationFuture,
+      builder: (context, snapshot) {
+        if (snapshot.connectionState == ConnectionState.done) {
+          return ApplePayButton(
+            paymentConfiguration: snapshot.data,
+            paymentItems: [
+              PaymentItem(
+                label: _merchantName,
+                amount: (widget.amount / 100).toStringAsFixed(2),
+              ),
+            ],
+            type: ApplePayButtonType.inStore,
+            onPaymentResult: _onApplePayResult,
+            onPressed: widget.onPressed,
+            width: MediaQuery.of(context).size.width,
+            height: 40,
+            onError: _onApplePayError,
+            loadingIndicator: const Center(
+              child: CircularProgressIndicator(),
+            ),
+          );
+        }
+        return const SizedBox.shrink();
+      },
     );
   }
+
+  Future<PaymentConfiguration> _getPaymentConfigurationFromAsset() {
+    return PaymentConfiguration.fromAsset(widget.paymentProfilePath);
+  }
```

Comment:
Previous comment:
> Why is it part of the library? If I understand correctly, these files are on the application side.
It looks like it's better to have PaymentConfiguration passed by the application, and it looks like in the original library this data is passed from the app

Addition:
It looks very strange that the library loads something from the assets of the application.
Let's do it this way: the application passes a `PaymentConfiguration` object to the library. This means that this code:
```
Future<PaymentConfiguration> _getPaymentConfigurationFromAsset() {
  return PaymentConfiguration.fromAsset(widget.paymentProfilePath);
}
```
should be at the application level. Not in the library.
Thus, the library will not load anything from the application's assets. The application will load it from the assets on its own. The library will receive the `PaymentConfiguration` object and work with it.

Or do we have some problem with that?